ultravox/data/datasets.py

```diff
@@ -1001,6 +1042,10 @@ def __init__(
                 for s in config.splits
             ]
         )
+        if not config.total_samples:
+            raise ValueError("total_samples must be set in the dataset config")
```

Comment:
If `total_samples` is required, then enforce it in the dataclass instead of raising an error here.